matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessorTest.kt

```diff
@@ -0,0 +1,387 @@
+            entity.isActive shouldBeEqualTo false
+        }
+        val workManager = fakeWorkManagerProvider.instance.workManager
+        verify { workManager.enqueueUniqueWork(any(), any(), any<OneTimeWorkRequest>()) }
+    }
+
+    @Test
+    fun `given beacon info and existing entity when beacon content is correct and inactive then it is aggregated`() {
+        val unsignedData = UnsignedData(
+                age = 123,
+                replacesState = AN_EVENT_ID
+        )
+        val event = Event(
+                senderId = A_SENDER_ID,
+                eventId = "",
+                unsignedData = unsignedData
+        )
+        val beaconInfo = MessageBeaconInfoContent(
+                isLive = false,
+                unstableTimestampMillis = A_TIMESTAMP,
+                timeout = A_TIMEOUT_MILLIS
+        )
+        fakeClock.givenEpoch(A_TIMESTAMP + 5000)
+        val aggregatedEntity = mockLiveLocationShareAggregatedSummaryEntityForEvent()
+        val previousEntities = mockPreviousLiveLocationShareAggregatedSummaryEntities()
+
+        val result = liveLocationAggregationProcessor.handleBeaconInfo(
+                realm = fakeRealm.instance,
+                event = event,
+                content = beaconInfo,
+                roomId = A_ROOM_ID,
+                isLocalEcho = false
+        )
+
+        result shouldBeEqualTo true
+        aggregatedEntity.eventId shouldBeEqualTo AN_EVENT_ID
+        aggregatedEntity.roomId shouldBeEqualTo A_ROOM_ID
+        aggregatedEntity.userId shouldBeEqualTo A_SENDER_ID
+        aggregatedEntity.isActive shouldBeEqualTo false
+        aggregatedEntity.endOfLiveTimestampMillis shouldBeEqualTo A_TIMESTAMP + A_TIMEOUT_MILLIS
+        aggregatedEntity.lastLocationContent shouldBeEqualTo null
+        previousEntities.forEach { entity ->
+            entity.isActive shouldBeEqualTo false
+        }
+        val workManager = fakeWorkManagerProvider.instance.workManager
+        verify { workManager.cancelUniqueWork(any()) }
+    }
+
+    @Test
+    fun `given beacon location data when it is local echo then it is ignored`() {
+        val event = Event(senderId = A_SENDER_ID)
+        val beaconLocationData = MessageBeaconLocationDataContent()
+
+        val result = liveLocationAggregationProcessor.handleBeaconLocationData(
+                realm = fakeRealm.instance,
+                event = event,
+                content = beaconLocationData,
+                roomId = A_ROOM_ID,
+                relatedEventId = AN_EVENT_ID,
+                isLocalEcho = true
+        )
+
+        result shouldBeEqualTo false
+    }
+
+    @Test
+    fun `given beacon location data when relatedEventId is null or empty then it is ignored`() {
+        val event = Event(senderId = A_SENDER_ID)
+        val beaconLocationData = MessageBeaconLocationDataContent()
+
+        val resultNoRelatedEventId = liveLocationAggregationProcessor.handleBeaconLocationData(
```

Comment:
as we're actioning different inputs `no related event id` and `empty related event id`, this feels like two separate tests (similar comment for the test below)

a parameterised test could potentially cover all these validation cases (_orrrrr_ the naive parameterised tests I've started doing [like this](https://github.com/vector-im/element-android/blob/develop/vector/src/test/java/im/vector/app/features/onboarding/RegistrationActionHandlerTest.kt#L48)

---

for the parameters themselves, are null/empty ids possible? would be nice if we could catch them further up the chain and avoid invalid empty ids

Reply:
I agree with your point. I created a loop on a list of ignored values to group the similar tests: see [912244a](https://github.com/vector-im/element-android/pull/6267/commits/912244a7648d09eea5c23dd595eddfaf5083dd3d). I didn't use parametrized tests since I don't like the boilerplate syntax in Junit 4. I am wondering if it is possible to switch to Junit 5 in Android: the syntax for parametrized tests is much nicer.

For you last point, I agree but I didn't change anything since it would require to change the events aggregation process in a more general way.